impeller/renderer/backend/vulkan/gpu_tracer_vk.cc

```diff
@@ -5,127 +5,155 @@
 #include "impeller/renderer/backend/vulkan/gpu_tracer_vk.h"
 
 #include <utility>
+#include "fml/logging.h"
 #include "fml/trace_event.h"
 #include "impeller/base/validation.h"
-#include "impeller/renderer/backend/vulkan/command_buffer_vk.h"
-#include "impeller/renderer/backend/vulkan/command_encoder_vk.h"
 #include "impeller/renderer/backend/vulkan/context_vk.h"
-#include "impeller/renderer/command_buffer.h"
+#include "vulkan/vulkan.hpp"
 
 namespace impeller {
 
-static constexpr uint32_t kPoolSize = 32u;
+static constexpr uint32_t kPoolSize = 64u;
 
-GPUTracerVK::GPUTracerVK(const std::weak_ptr<ContextVK>& context)
-    : context_(context) {
-  auto strong_context = context_.lock();
-  if (!strong_context) {
-    return;
-  }
-  timestamp_period_ = strong_context->GetPhysicalDevice()
+GPUTracerVK::GPUTracerVK(const std::shared_ptr<DeviceHolder>& device_holder)
+    : device_holder_(device_holder) {
+  timestamp_period_ = device_holder_->GetPhysicalDevice()
                           .getProperties()
                           .limits.timestampPeriod;
   if (timestamp_period_ <= 0) {
     // The device does not support timestamp queries.
     return;
   }
-  vk::QueryPoolCreateInfo info;
-  info.queryCount = kPoolSize;
-  info.queryType = vk::QueryType::eTimestamp;
-
-  auto [status, pool] = strong_context->GetDevice().createQueryPoolUnique(info);
-  if (status != vk::Result::eSuccess) {
-    VALIDATION_LOG << "Failed to create query pool.";
-    return;
-  }
-  query_pool_ = std::move(pool);
   // Disable tracing in release mode.
 #ifdef IMPELLER_DEBUG
   valid_ = true;
 #endif
 }
 
-void GPUTracerVK::RecordStartFrameTime() {
+bool GPUTracerVK::IsValid() const {
```

Comment:
Just being pedantic, `IsValid` feels weird given it's totally valid for this instance to exist in an invalid state (i.e. either timestamp_period not being supported, or debug mode being disabled). 

Consider renaming this `IsEnabled()` and consider making it private since there isn't (AFAICT) a reason for another class to check this property, it's internal-only state.

Reply:
This is more or less just for testing. Update to IsEnabled SGTM

Reply:
Done